crates/rpc/src/server2.rs

```diff
@@ -799,3 +716,31 @@ where
         self.project().send_stream.poll_close(cx).map_err(Error::io)
     }
 }
+
+impl<RPC: RpcV2> ResponseSink<RPC> {
+    /// Waits until this [`ResponseSink`] is closed.
+    pub async fn wait_closed(&mut self) {
+        self.send_stream.get_mut().stopped().map(drop).await
+    }
+
+    /// Sends all RPC responses from the provided [`Stream`] into this
+    /// [`ResponseSink`].
+    pub async fn send_all(&mut self, stream: impl Stream<Item = RPC::Response>) -> Result<()> {
```

Comment:
If the method is on a sink now, I'd call it something like `forward()` maybe, for consistency with the [`futures` API](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.forward), though I do realize that the actual `forward()` method is on a `Stream` there.